homeassistant/components/azure_data_explorer/__init__.py

```diff
@@ -0,0 +1,225 @@
+from .const import (
+    CONF_FILTER,
+    CONF_SEND_INTERVAL,
+    DATA_FILTER,
+    DATA_HUB,
+    DEFAULT_MAX_DELAY,
+    DOMAIN,
+    FILTER_STATES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional(CONF_FILTER, default={}): FILTER_SCHEMA,
+            },
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, yaml_config: ConfigType) -> bool:
+    """Activate ADX component from yaml.
+
+    Adds an empty filter to hass data.
+    Tries to get a filter from yaml, if present set to hass data.
+    If config is empty after getting the filter, return, otherwise emit
+    deprecated warning and pass the rest to the config flow.
+    """
+
+    hass.data.setdefault(DOMAIN, {DATA_FILTER: FILTER_SCHEMA({})})
+    if DOMAIN not in yaml_config:
+        return True
+    hass.data[DOMAIN][DATA_FILTER] = yaml_config[DOMAIN][CONF_FILTER]
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Do the setup based on the config entry and the filter from yaml."""
+    hass.data.setdefault(DOMAIN, {DATA_FILTER: FILTER_SCHEMA({})})
+    adx = AzureDataExplorer(hass, entry)
+    try:
+        await adx.test_connection()
+    except KustoServiceError as exp:
+        raise ConfigEntryError(
+            "Could not find Azure Data Explorer database or table"
+        ) from exp
+    except KustoAuthenticationError:
+        return False
+
+    hass.data[DOMAIN][DATA_HUB] = adx
+    entry.async_on_unload(entry.add_update_listener(async_update_listener))
+    await adx.async_start()
+    return True
+
+
+async def async_update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Update listener for options."""
+    hass.data[DOMAIN][DATA_HUB].update_options(entry.options)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    adx = hass.data[DOMAIN].pop(DATA_HUB)
+    await adx.async_stop()
+    return True
+
+
+class AzureDataExplorer:
+    """A event handler class for Azure Data Explorer."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the listener."""
+
+        self.hass = hass
+        self._entry = entry
+        self._entities_filter = hass.data[DOMAIN][DATA_FILTER]
+
+        self._client = AzureDataExplorerClient(
+            CONF_ADX_CLUSTER_INGEST_URI=self._entry.data["clusteringesturi"],
+            CONF_ADX_DATABASE_NAME=self._entry.data["database"],
+            CONF_ADX_TABLE_NAME=self._entry.data["table"],
+            CONF_APP_REG_ID=self._entry.data["client_id"],
+            CONF_APP_REG_SECRET=self._entry.data["client_secret"],
+            CONF_AUTHORITY_ID=self._entry.data["authority_id"],
+            CONF_USE_FREE=self._entry.data["use_free_cluster"],
+        )
```

Comment:
```suggestion
        self._client = AzureDataExplorerClient(**entry.data)
```

Reply:
I was ask earlier to write this out for readability. When applying the the suggested change to code, it breaks

Reply:
Then at least use the constant like `CONF_USE_FREE`

Reply:
```suggestion
        self._client = AzureDataExplorerClient(entry.data)
```